src/shared/nsflags.c

```diff
@@ -40,18 +41,40 @@ int namespace_flags_from_string(const char *name, unsigned long *ret) {
 }
 
 int namespace_flags_to_string(unsigned long flags, char **ret) {
+        _cleanup_strv_free_ char **l = NULL;
         _cleanup_free_ char *s = NULL;
+        int r;
+
+        r = namespace_flags_to_strv(flags, &l);
+        if (r < 0)
+                return r;
+
+        s = strv_join(l, NULL);
+        if (!s)
+                return -ENOMEM;
+
+        if (ret)
+                *ret = TAKE_PTR(s);
```

Comment:
Please also add `assert(ret)` to the start of this function instead of making it optional.